.changeset/poor-glasses-vanish.md

```diff
@@ -0,0 +1,11 @@
+---
+"@graphql-tools/federation": patch
+"@graphql-tools/stitch": patch
+---
+
+When the gateway receives the query, now it chooses the best root field if there is the same root field in different subgraphs.
+For example, if there is `node(id: ID!): Node` in all subgraphs but one implements `User` and the other implements `Post`, the gateway will choose the subgraph that implements `User` or `Post` based on the query.
+
+If there is a unresolvable interface field, it throws.
+
+See [this supergraph and the test query](https://github.com/ardatan/graphql-tools/pull/6141/files#diff-81b2b8ce59ac79aa15bc63735fbc6edc6551fbe46b39468fa81199825aa2460c) to see a real-life example
```

Comment:
Better point to actual code in branch instead of a PR changed files diff.

Reply:
I pointed to the example directly!

Reply:
Let me know if it is better now

Reply:
Actually, this branch will be deleted after merging. Maybe use a permalink, or update the changelog to point to master branch once merged.

Reply:
I can actually point to the master branch now